echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+    """
+   Compute the overall weight proportions for aged and uanged fish within each stratum 
+
+    Parameters
+    ----------
+    specimen_data: pd.DataFrame
+        Dataframe containing specimen weights
+    weight_strata: pd.DataFrame
+        Dataframe contained summed weights of both aged and unaged fish
+    """   
+    ### Calculate adult:all age ratio for each stratum
+    # ---- Drop unaged fish
+    specimen_data_filtered = specimen_data[ specimen_data.sex != 'unsexed' ].dropna( how = 'any' , subset = 'age' ) 
+
+    # ---- Sum the weight of adult fish
+    specimen_adult = specimen_data_filtered[ specimen_data_filtered.age > 1 ].groupby( [ 'stratum_num' ] )[ 'weight' ].sum( )
+
+    # ---- Sum the weight of all fish
+    specimen_all = specimen_data_filtered.groupby( [ 'stratum_num' ] )[ 'weight' ].sum( ).to_frame( )
+
+    # ---- Calculate the ratio between the two groups
+    specimen_all[ 'weight_ratio' ] = ( specimen_adult / specimen_all.weight ).fillna( 0 )
+
+    ### Calculate the aged proportions of all fish
+    # ---- Merge with `weight_strata`
+    aged_proportions = specimen_all.reset_index( ).merge( weight_strata , on = [ 'stratum_num' ] )
+
+    # ---- Calculate the aged proportion of all fish
+    aged_proportions[ 'proportion_aged_weight_all' ] = (
+        aged_proportions.weight / aged_proportions.weight_stratum_all
+    )
+
+    # ---- Calculate the aged proportion of adult fish
+    aged_proportions[ 'proportion_aged_weight_adult' ] = (
+        aged_proportions.proportion_aged_weight_all * aged_proportions.weight_ratio
+    )
+
+    ### Calculate the unaged weight proportions
+    # ---- All fish
+    aged_proportions[ 'proportion_unaged_weight_all' ] = (
+        1.0 - aged_proportions.proportion_aged_weight_all
+    )
+
+    # ---- Adult fish
+    aged_proportions[ 'proportion_unaged_weight_adult' ] = (
+        1.0 - aged_proportions.proportion_aged_weight_adult
+    )
+
+    return aged_proportions.drop( [ 'weight_ratio' ] , axis = 1 )
+
+def compute_aged_weight_proportions( specimen_data: pd.DataFrame ,
+                                     length_intervals: np.ndarray ,
+                                     age_intervals: np.ndarray ):
+    """
+    Calculate length-age binned weight proportions    
+
+    Parameters
+    ----------
+    specimen_data: pd.DataFrame
+        Dataframe containing specimen weights
+    length_intervals: np.ndarray
+        Array containing length bins/intervals
+    age_intervals: np.ndarray
+        Array containing age bins/intervals
+    """ 
+
+    ### Process the specimen data 
+    # ---- Drop unaged fish
+    # ==== !!! TODO: pending what FEAT says, weights associated with 
+    # ==== missing ages should be added into the 'unaged' category. 
+    # ==== This would further mean apportioning these into `weight_strata_aged_uanged`
+    specimen_data_filtered = specimen_data[ specimen_data.sex != 'unsexed' ].dropna( how = 'any' , subset = 'age' )
```

Comment:
This does 2 things: select only sexed fish and dropped unaged fish, right?

Are there other place where unsexed fish from station 1 is used? If not, maybe we could also drop unsexed fish from station 1 when `specimen_df` is first read in. Or is it that there are too many unsexed fish so it is not a good idea to just drop them at data loader?

Reply:
- Drop unaged fish in data loader
- Adding caveat doc page related to the treatment of unsexed and removal of unaged fish
 - Which functions do the unsexed fish become relevant (e.g. length-weight regression)
- Print out in console in data loader: "x unaged samples removed from Station 2 dataset" or something to that effect